src/mono/mono/mini/CMakeLists.txt

```diff
@@ -79,6 +79,13 @@ endif()
   addprefix(icu_shim_sources "${ICU_SHIM_PATH}" "${icu_shim_sources_base}")
 
   if (TARGET_DARWIN)
+    if(CLR_CMAKE_TARGET_IOS)
```

Comment:
Why are these needed ? We define these in mono/CMakeLists.txt.

Reply:
`xamarin-macios` build is failing after my PR https://github.com/dotnet/runtime/pull/93220 , where I have checks like https://github.com/dotnet/runtime/blob/main/src/native/libs/System.Globalization.Native/entrypoints.c#L22 , targets are not defined, my guess was that these block https://github.com/dotnet/runtime/blob/main/src/mono/CMakeLists.txt#L751-L756 doesn't work properly.

Reply:
HOST_IOS should be defined at:
https://github.com/dotnet/runtime/blob/main/src/mono/CMakeLists.txt#L212
If that is not executed, many things wouldn't work.

Reply:
I see in that case this will not be needed.